include/envoy/network/connection.h

```diff
@@ -120,14 +120,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
-
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
   /**
    * @return the local address of the connection. For client connections, this is the origin
    * address. For server connections, this is the local destination address. For server connections
    * it can be different from the proxy address if the downstream connection has been redirected or
    * the proxy is operating in transparent mode.
    */
   virtual const Address::Instance& localAddress() PURE;
+  virtual Address::InstanceConstSharedPtr localAddressConstSharedPtr() PURE;
```

Comment:
So the basic motivation here is to form a `CidrRange` without having to copy the underlying Address instance, as response to my review comment that we shouldn't have to convert an Address instance to string and then reparse to form an `Address:InstanceConstSharedPtr`. The reason we need to form the `CidrRange` is for some inclusion testing. Does that sound right?

It seems reasonable to me given the current way in which addresses are handled. From a pure performance PoV, I wonder if the inclusion testing could not depend on a const shared pointer in the first place? I.e. `isInRange()` shouldn't need to do this construction of `CidrRange` to begin with. That way you save on the reference count increment/decrement.

@jamessynge @mattklein123